web/src/main/java/com/epam/esm/web/advice/CustomAdvice.java

```diff
@@ -0,0 +1,88 @@
+
+import com.epam.esm.service.exeption.IllegalRequestParameterException;
+import com.epam.esm.service.exeption.IllegalRequestSortParameterException;
+import com.epam.esm.service.exeption.RecourseExistException;
+import com.epam.esm.service.exeption.RecourseNotExistException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.sql.SQLException;
+import java.time.LocalDateTime;
+import java.util.Locale;
+
+@ControllerAdvice
+public class CustomAdvice extends ResponseEntityExceptionHandler {
+    @ExceptionHandler(RecourseExistException.class)
+    public ResponseEntity<ErrorResponse> handleRecourseExistException(RecourseExistException e) {
+        ErrorResponse response = new ErrorResponse();
+        response.setMessage(e.getMessage());
+        response.setLocalDateTime(LocalDateTime.now());
+        String str=CustomErrorCode.RECOURSE_EXIST.getStatusCode();
+        response.setStatusCode(CustomErrorCode.RECOURSE_EXIST.getStatusCode());
+        return new ResponseEntity<>(response, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @ExceptionHandler(RecourseNotExistException.class)
+    public ResponseEntity<ErrorResponse> handleRecourseNotExistException(RecourseNotExistException e) {
+        ErrorResponse response = new ErrorResponse();
+        response.setMessage(e.getMessage());
+        response.setLocalDateTime(LocalDateTime.now());
+        response.setStatusCode(CustomErrorCode.RECOURSE_NOT_EXIST.getStatusCode());
+        return new ResponseEntity<>(response, HttpStatus.NOT_FOUND);
+    }
+    @ExceptionHandler(IllegalRequestSortParameterException.class)
+    public ResponseEntity<ErrorResponse> handleIllegalRequestSortParameterException(IllegalRequestSortParameterException e) {
+        ErrorResponse response = new ErrorResponse();
+        response.setMessage(e.getMessage());
+        response.setLocalDateTime(LocalDateTime.now());
+        response.setStatusCode(CustomErrorCode.ILLEGAL_SORT_PARAMETER.getStatusCode());
+        return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
+    }
+    @ExceptionHandler(SQLException.class)
+    public ResponseEntity<ErrorResponse> handleIllegalRequestSortParameterException(SQLException e) {
+        ErrorResponse response = new ErrorResponse();
+        response.setMessage(e.getMessage());
+        response.setLocalDateTime(LocalDateTime.now());
+        response.setStatusCode(CustomErrorCode.ILLEGAL_SORT_PARAMETER.getStatusCode());
```

Comment:
all the sqlExceptions are considered as ILLEGAL_SORT_PARAMETER?

Reply:
done